src/distilabel/distiset.py

```diff
@@ -13,21 +13,31 @@
 # limitations under the License.
 
 import logging
+import os.path as posixpath
 import re
+from os import PathLike
 from pathlib import Path
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Union
 
-from datasets import load_dataset
+import fsspec
+import yaml
+from datasets import Dataset, load_dataset, load_from_disk
+from datasets.filesystems import is_remote_filesystem
 from huggingface_hub import DatasetCardData, HfApi
+from huggingface_hub.file_download import hf_hub_download
 from pyarrow.lib import ArrowInvalid
+from typing_extensions import Self
 
-from distilabel.steps.tasks.base import DISTILABEL_METADATA_KEY
 from distilabel.utils.card.dataset_card import (
     DistilabelDatasetCard,
     size_categories_parser,
 )
 from distilabel.utils.files import list_files_in_dir
 
+DISTISET_CONFIG_FOLDER: str = "distiset_configs"
+PIPELINE_CONFIG_FILENAME: str = "pipeline.yaml"
+PIPELINE_LOG_FILENAME: str = "pipeline.log"
```

Comment:
if we want to be a bit more precise with the type hint we can use `typing.Final[str]`